webapps-schema/src/main/java/io/camunda/webapps/schema/entities/usermanagement/EntityJoinRelation.java

```diff
@@ -7,40 +7,43 @@
  */
 package io.camunda.webapps.schema.entities.usermanagement;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
 public record EntityJoinRelation(String name, Long parent) {
 
+  @JsonIgnore
+  public String getRouting() {
+    return parent != null ? parent.toString() : null;
+  }
+
   public static class EntityJoinRelationFactory {
 
-    private final IdentityJoinRelationshipType parent;
-    private final IdentityJoinRelationshipType child;
+    private final String parentName;
+    private final String childName;
 
-    public EntityJoinRelationFactory(
-        final IdentityJoinRelationshipType parent, final IdentityJoinRelationshipType child) {
-      this.parent = parent;
-      this.child = child;
+    public EntityJoinRelationFactory(final String parentName, final String childName) {
+      this.parentName = parentName;
+      this.childName = childName;
     }
 
     public EntityJoinRelation createParent() {
-      return new EntityJoinRelation(parent.getType(), null);
+      return new EntityJoinRelation(parentName, null);
     }
 
     public EntityJoinRelation createChild(final long parentKey) {
-      return new EntityJoinRelation(child.getType(), parentKey);
+      return new EntityJoinRelation(childName, parentKey);
     }
-  }
-
-  public enum IdentityJoinRelationshipType {
-    GROUP("group"),
-    MEMBER("member");
 
-    private final String type;
+    public String getChildName() {
+      return childName;
+    }
 
-    IdentityJoinRelationshipType(final String type) {
-      this.type = type;
+    public String getChildKey(final long parentKey, final long childKey) {
+      return "%s-%s/%s-%s".formatted(parentName, childName, parentKey, childKey);
```

Comment:
I believe the naming makes this a bit more confusing that it needs to be. I've been thinking of renaming `EntityJoinRelationFactory` to `EntityJoinRelation`, make it a top-level class and then move the old `EntityJoinRelation` inside as `EntityJoinRelation.JoinValue`. That might make it clearer that the "factory" models the relation and the value is what gets serialized in ES. What do you think?

Reply:
❌ Since you're heavily refactoring the `EntityJoinRelation` code, let's move the whole refactoring into a separate PR and create a GitHub issue for it that lists the reasoning behind it.

We're putting many refactoring changes into a PR intended to introduce handlers for Role membership addition and removal. It doesn't make sense, and it will be confusing as "code documentation" for anyone who tries to understand these changes in the future.

I don't mind the refactoring, but I feel that we're investing too much time (that we don't have) into optimizing this code.

Reply:
These seemed like fairly small changes to me. I generally try to make code better as I touch it, so I often refactor details as I go. I didn't spend much time on it, and I believe it will save time as we keep reusing the code for the remaining many to many relationships.
I can move the refactoring to a separate PR if you prefer.

Reply:
@koevskinikola I think I reverted all controversial changes. Does this look better now?

Reply:
Yes, thank you! 

Feel free to ping me for the refactoring PR, I'd be happy to help with the refactoring and review.